### PR TITLE
TS-4749: generate warning message when origin_max_connections limits

### PR DESCRIPTION
**Pull request by shinrich**

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/419/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/522/ for details.

**Comment by zwoop** [comment]
Clang-format.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/529/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/426/ for details.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -4865,12 +4865,13 @@ HttpSM::do_http_server_open(bool raw)
...
                            strlen(t_state.current.server->name));
 
-    ip_port_text_buffer addrbuf;
     if (connections->getCount(t_state.current.server->dst_addr, hostname_hash,
                               (TSServerSessionSharingMatchType)t_state.txn_conf->server_session_sharing_match) >=
         t_state.txn_conf->origin_max_connections) {
-      DebugSM("http", "[%" PRId64 "] over the number of connection for this host: %s", sm_id,
-              ats_ip_nptop(&t_state.current.server->dst_addr.sa, addrbuf, sizeof(addrbuf)));
+      ip_port_text_buffer addrbuf;
+      ats_ip_ntop(&t_state.current.server->dst_addr.sa, addrbuf, sizeof(addrbuf));
```
Comment (bryancall) [inline proxy/http/HttpSM.cc]: You would want to call `ats_ip_nptop`.  `ats_ip_ntop` will only copy the address.

**Comment by shinrich** [comment]
Addressed @bryancall 's comment via another squashed commit.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/459/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/563/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/564/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/461/ for details.

**Pull request merged**